src/components/truncated-text.tsx

```diff
@@ -0,0 +1,6 @@
+import React, { FC } from 'react';
+import { Text, TextProps } from 'react-native';
+
+export const TruncatedText: FC<TextProps> = props => (
+  <Text numberOfLines={1} ellipsizeMode="tail" {...props} style={[props.style, { flexShrink: 1 }]} />
```

Comment:
Please consider extracting some values from `props` and using the rest props to avoid spreading props in the middle